api/v1alpha1/capiprovider_types.go

```diff
@@ -104,19 +105,23 @@ type WorkloadIdentityRef struct {
 type CAPIProviderStatus struct {
 	// Indicates the provider status
 	// +kubebuilder:default=Pending
-	State ProviderState `json:"state,omitempty"`
+	Phase Phase `json:"phase,omitempty"`
 
 	// Variables is a map of environment variables added to the content of the ConfigSecret
 	// +kubebuilder:default={CLUSTER_TOPOLOGY:"true",EXP_CLUSTER_RESOURCE_SET:"true",EXP_MACHINE_POOL: "true"}
 	Variables map[string]string `json:"variables,omitempty"`
 
-	ProviderStatus *operatorv1.ProviderStatus `json:",inline"`
+	operatorv1.ProviderStatus `json:",inline"`
 }
 
-//+kubebuilder:object:root=true
-//+kubebuilder:subresource:status
-
 // CAPIProvider is the Schema for the CAPI Providers API.
+//
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Type",type="string",JSONPath=".spec.type"
+// +kubebuilder:printcolumn:name="ProviderName",type="string",JSONPath=".spec.name"
+// +kubebuilder:printcolumn:name="InstalledVersion",type="string",JSONPath=".status.installedVersion"
```

Comment:
Q: do we have `.installedVersion` field defined in the current API?

Reply:
Yes, this is a part of inlined status of the CAPI operator